src/util/segmentation/operations/correction.js

```diff
@@ -0,0 +1,524 @@
+import { fillInside, eraseInside } from './index.js';
+import getPixelPathBetweenPixels from './getPixelPathBetweenPixels';
+import clip from '../../clip';
+import pointInImage from '../../pointInImage';
+
+import { getLogger } from '../../logger';
+import floodFill from './floodFill.js';
+
+const logger = getLogger('util:segmentation:operations:correction');
+
+/**
+ * correction - Using the stroke given, determine which action(s) to perfom:
+ * - Stroke starts and ends inside a segmentation: Behaves as an subtractive freehand scissors.
+ * - Stroke starts and ends outside a segmentation: Behaves as an additive freehand scissors.
+ * - Stroke out-in-out: Section is subtracted.
+ * - Stroke in-out-in: Section is added.
+ *
+ * @param  {Object} evt The cornerstone event.
+ * @returns {null}
+ */
+export default function correction(evt) {
+  const { operationData } = evt;
```

Comment:
`const { segmentationMixinType, pixelData, segmentIndex } = evt.operationData;`

Reply:
👍 Applied throughout, and generally cleaned all the operations up a bit.